transport/grpc/header.go

```diff
@@ -0,0 +1,85 @@
+package grpc
+
+import (
+	"strings"
+
+	"github.com/yarpc/yarpc-go/transport"
+	"google.golang.org/grpc/metadata"
+)
+
+// headerMapper converts gRCP Metadata to and from transport headers.
+type headerMapper struct{ Prefix string }
+
+var (
+	applicationHeaders = headerMapper{ApplicationHeaderPrefix}
+)
+
+// toGRPCMetadata converts transport headers into gRPC Metadata.
+//
+// Headers are read from 'from' and written to 'to'. The final header collection
+// is returned.
+//
+// If 'to' is nil, a new map will be assigned.
+func (hm headerMapper) ToGRPCMetadata(from transport.Headers, to metadata.MD) metadata.MD {
```

Comment:
No need to name this `ToGRPCMetadata`. Just `ToMetadata` should be fine.

Also, don't make this public. This shouldn't be part of our public API. Move it to an internal package if necessary.